code/esp_code/tcp_client.c

```diff
@@ -0,0 +1,217 @@
+/* BSD Socket API Example
+
+   This example code is in the Public Domain (or CC0 licensed, at your option.)
+
+   Unless required by applicable law or agreed to in writing, this
+   software is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+   CONDITIONS OF ANY KIND, either express or implied.
+*/
+#include <stddef.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/param.h>
+
+#include "freertos/FreeRTOS.h"
+#include "freertos/event_groups.h"
+#include "freertos/queue.h"
+#include "freertos/semphr.h"
+#include "freertos/task.h"
+
+#include "lwip/dns.h"
+#include "lwip/err.h"
+#include "lwip/igmp.h"
+#include "lwip/sockets.h"
+#include "lwip/sys.h"
+#include <lwip/netdb.h>
+
+#include "driver/gpio.h"
+#include "driver/spi_slave.h"
+#include "esp32/rom/cache.h"
+#include "esp_event.h"
+#include "esp_event_loop.h"
+#include "esp_log.h"
+#include "esp_spi_flash.h"
+#include "esp_system.h"
+#include "esp_wifi.h"
+#include "nvs_flash.h"
+#include "protocol_examples_common.h"
+#include "soc/rtc_periph.h"
+#include "tcpip_adapter.h"
+
+#ifdef CONFIG_EXAMPLE_IPV4
+#define HOST_IP_ADDR CONFIG_EXAMPLE_IPV4_ADDR
+#else
+#define HOST_IP_ADDR CONFIG_EXAMPLE_IPV6_ADDR
+#endif
+
+#define PORT CONFIG_EXAMPLE_PORT
+
+// Pins in use. The SPI Master can use the GPIO mux, so feel free to change
+// these if needed.
+#define GPIO_HANDSHAKE 15
+#define GPIO_MOSI 23
+#define GPIO_MISO 19
+#define GPIO_SCLK 18
+#define GPIO_CS 2
+
+static const char *TAG = "example";
+// static const char *payload = "Message from ESP32 ";
+
+size_t buffer_size = 129;
+
+// Called after a transaction is queued and ready for pickup by master. We use
+// this to set the handshake line high.
+void my_post_setup_cb(spi_slave_transaction_t *trans) {
+    WRITE_PERI_REG(GPIO_OUT_W1TS_REG, (1 << GPIO_HANDSHAKE));
+}
+
+// Called after transaction is sent/received. We use this to set the handshake
+// line low.
+void my_post_trans_cb(spi_slave_transaction_t *trans) {
+    WRITE_PERI_REG(GPIO_OUT_W1TC_REG, (1 << GPIO_HANDSHAKE));
+}
+
+// buffers for spi
+DMA_ATTR char sendbuf[129] = "";
+DMA_ATTR char recvbuf[129] = "";
+
+static void tcp_client_task(void *pvParameters) {
+    // char rx_buffer[128];
+    char addr_str[128];
+    int addr_family;
+    int ip_protocol;
+
+    while (1) {
+
+#ifdef CONFIG_EXAMPLE_IPV4
```

Comment:
We're not going to use ipv6, so you can remove these blocks from the code.

Reply:
I will, but do you mind if I create an issue and fix this next iteration?

Reply:
I don't think this example/dead code should be in master

Reply:
I removed it anyways in #4